### ISSUE TITLE
Running procedure with output parameters using shortcut method

### ISSUE THREAD

**Issue by enghelewa** [yajra/laravel-oci8#337]
### Summary of problem or feature request
When executing procedure which has an Out parameter using the shortcut method you get this error:
` ORA-06502: PL/SQL: character string buffer too small`
According to [PDO Docs](http://php.net/manual/en/pdostatement.bindparam.php) 

> To indicate that a parameter is an OUT parameter from a stored procedure, you must explicitly set the length.

while the `executeProcedure` passes nothing to `bindParam` other than a key value pairs.

```
public function executeProcedure($procedureName, array $bindings = [])
    {
        $stmt = $this->createStatementFromProcedure($procedureName, $bindings);

        foreach ($bindings as $key => &$value) {
            $stmt->bindParam(':' . $key, $value);
        }

        return $stmt->execute();
    }
```

- Operating System = Windows 10
- PHP Version = 7.1.9
- Laravel Version = 5.5
- Laravel-OCI8 Version = 5.5

**Comment by mstaack** [comment]
@enghelewa please show me your procedure defintion. are you return and OUT Cursor?

Cheers Max

**Comment by enghelewa** [comment]
This is my test package
```
CREATE OR REPLACE package body test_pkp is
  PROCEDURE get_x (x IN NUMBER,y OUT NUMBER)
    IS 
    BEGIN
       y:=x;
    END get_x;
end test_pkp;
```

and this is my test call in laravel
```
$procedureName = 'TEST_PKP.GET_X';
$p2=0;
$bindings = [
   'p1'  => 10,
   'p2' => $p2
];
DB::executeProcedure($procedureName, $bindings);
dd($p2);
```

**Comment by enghelewa** [comment]
This will work very will. But the shortcut way is way better and less code to write
```
$pdo = DB::getPdo();
$p1 = 10;
$p2 = 10;
$stmt = $pdo->prepare("begin TEST_PKP.GET_X(:p1, :p2); end;");
```
then this will work as **_`:p1`_** is a **_`IN`_** field.
```
$stmt->bindParam(':p1', $p1);   //<---------------
$stmt->bindParam(':p2', $p2,PDO::PARAM_INT);
$stmt->execute();
dd($p2);
```
but this will not work as **_`:p2`_** is a **_`OUT`_** field.
```
$stmt->bindParam(':p1', $p1);
$stmt->bindParam(':p2', $p2);     //<---------------
$stmt->execute();
dd($p2);
```

**Comment by mstaack** [comment]
@enghelewa yeah sounds reasonable...

to create your statement you could use `createStatementFromProcedure($procedureName, $bindings)`

do your custom bindings and execute the statement.

but sure.... this is not really nice.. how about a 3rd parameter with binding types?

i also tried something like this:
```
        $input  = 20;
        $output = 0;

        $bindings = [
            'p1' => $input,
            'p2' => [
                'value' => &$output,
                'type'  => PDO::PARAM_INT
            ]
        ];
```  

and

```
        foreach ($bindings as $key => &$binding) {
            $type = PDO::PARAM_STR;
            if (is_array($binding)) {
                $value = &$binding['value'];
                $type  = $binding['type'];
            }
            $stmt->bindParam(':'.$key, $value, $type);
        }
```

but couldn't get it to work. @yajra  @enghelewa any infos/help on handling type and value by reference?

cheers max

**Comment by enghelewa** [comment]
@mstaack  your code works but you are missing the _`else`_ statement
the input parameter still not bound that's why it returns null
```
if (is_array($binding)) {
	$value = &$binding['value'];
	$type = $binding['type'];
} else
	$value = $binding;
```

**Comment by mstaack** [comment]
@enghelewa https://github.com/yajra/laravel-oci8/pull/340

**Comment by enghelewa** [comment]
@mstaack thank you

**Comment by yajra** [comment]
Patch released on v5.4.18, kindly update and confirm. Thanks!

**Comment by yajra** [comment]
Release on v5.5.1 too via https://github.com/yajra/laravel-oci8/pull/344.
